handbook/direction/code-graph/code-insights/index.md

```diff
@@ -1,44 +1,218 @@
+- Latest demo – **TODO check with marketing if we're allowed to share a demo here**
+- [Documentation](https://docs.sourcegraph.com/code_insights)
 
-### 1. Deliver a user-loved, revenue-capturing product
+## Mission, Vision, and Guiding Principles
 
-#### 1.1 Deliver code insights beta version
+### Mission
 
-**Problem:** Code insights can't be [in beta](https://docs.sourcegraph.com/admin/beta_and_prototype_features) until:
+Code Insights helps engineering leaders, managers, and engineers answer questions about how their code is changing or what's in their code that are difficult or impossible to answer otherwise.
 
-1. Users can create insights effectively and without error
-   1. This includes an easy path to finding/creating insights, like through the search UI (one-click create from search)
-1. Code insights can run on the backend and meet a user's loading-time expectations
-   1. We'll build both backend optimizations and UI expectation-setting and communicating
-   1. This also involves exploring parallel rather than serial implementations
-1. We have docs for both of the above features
+We bring **the best practices of big data to challenges of big code**, turning a codebase into a queryable database.
 
-**Why we are focusing on this:** a beta that sets/meets appropriate user expectations, scales to their entire codebase, and is usable without hand-holding from CE/product is necessary to then ultimately enable insights by default in an open beta. An open beta in turn will allow us to collect feedback and add necessary features as we develop code insights into a revenue-capturing, generally available product.
+**We help engineering leaders** set and measure quantitative engineering goals, whether they be related to a migration, deprecation, vulnerability, code smell, or high level understanding. We also help engineering leaders understand the shape of their code and how it changes over time. Then we help them tie trends and metrics to _owners_ on their engineering teams.
 
-**How we will measure success:** We will track quantitative customer adoption and usage for metrics like number of insights created, number of users who have created an insight, and number of pageviews on code insights in our [Code Insights KPIs](https://sourcegraph.looker.com/dashboards/172) dashboard. We will also be in frequent contact with code insights customers and the CE team to collect qualitative feedback about the product.
+**We help individual engineers** measure the day-to-day goals of their teams: how they are progressing on an individual initiative, or how quickly other teams are adopting their code. We also help them get a historical usage view of functions, changes, or other patterns. Happily, we've heard that Code Insights has even helped some engineers get promotions or bonuses, because they use code insights to quantify the impact and spread of their work.
```

Comment:
Maybe we should also mention in a sentence here the value of making things that are obviously visible to engineers "down in the trenches" visible to their engineering leaders, by allowing them to create dashboards with visualizations for problems they are seeing and sharing them with their leadership that their management often doesn't notice. Many engineers struggle to get resources dedicated to paying down tech debt, increasing test coverage, etc. Leadership personas tend to look at graphs, but for business KPIs, not aware of the technical costs and state. Especially powerful if we can eventually overlay those two together to show correlations.

Reply:
good call, will add!